File: 3300-minimum-element-after-replacement-with-digit-sum/3300-minimum-element-after-replacement-with-digit-sum.cpp
class Solution {
public:
    int minElement(vector<int>& nums) {
        int n=nums.size();
        int ans=INT_MAX;
        for(int i=0;i<n;i++){
            int a=nums[i],b=nums[i],sum=0;
            while(a){
                 sum+=a%10;
                 a=a/10;
            }
        sum=min(sum,b);
        ans=min(ans,sum);
        }
        return ans;
    }
};

File: 3300-minimum-element-after-replacement-with-digit-sum/3300-minimum-element-after-replacement-with-digit-sum.java
class Solution {
    public int minElement(int[] nums) {
        
        for(int i=0; i<nums.length; i++){
            int sum=0;
            while(nums[i] > 0){
                int temp = nums[i];
                temp %= 10;
                sum += temp;
                nums[i] /= 10;
            }
            nums[i] = sum;
        }

        Arrays.sort(nums);
        return nums[0];
    }
}